### PR TITLE
Make session storage optional for merchant custom apps

### PR DESCRIPTION
**Pull request by lizkenyon**
(cherry picked from commit 9bf0e069a4e1b4c33359934aa0f451345756cb36)

### WHY are these changes introduced?

Continuation of [Improvements for custom Remix apps](https://github.com/Shopify/first-party-library-planning/issues/586)

Related [PR](https://github.com/Shopify/shopify-app-js/pull/1085)

### How was this tested
I tested that using this branch you are able to remove session storage, and authenticate a webhook request correctly for a merchant custom app.

### WHAT is this pull request doing?
* For merchant custom apps when we need a session we build it out of the configured access tokens provided in the `shopifyApp` config.

* This PR adds a helper function to either load the session from storage or in the case of merchant custom apps to create it from the configured tokens.

## Type of change

- [ ] Patch: Bug (non-breaking change which fixes an issue)
- [x] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ x I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [x] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

**Review by paulomarg — APPROVED** [review]
LGTM! I had nits which were mostly around function names, but approving so you can merge.

File: packages/apps/shopify-app-remix/src/server/authenticate/helpers/load-session.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import {AppDistribution, BasicParams} from '../../types';
+
+export async function loadSession(
```
[the remark this answers is not in the document]
Comment (paulomarg) [reply packages/apps/shopify-app-remix/src/server/authenticate/helpers/load-session.ts:3]: That could work, but it still feels a bit too generic? Naming things is hard.

[the remark this answers is not in the document]
Comment (lizkenyon) [reply packages/apps/shopify-app-remix/src/server/authenticate/helpers/load-session.ts:3]: `createOrLoadOfflineSession`

[the remark this answers is not in the document]
Comment (paulomarg) [reply packages/apps/shopify-app-remix/src/server/authenticate/helpers/load-session.ts]: I still kind of miss the reason behind this helper, which is a little more specific than the others. I'm thinking something like `getBaseSessionForShop` or some such, you know?

**Pull request merged**